File: ts/project/.sdk/src/cmp/zig/fragment/EntityCreateOp.fragment.zig
// EJECT-START

pub fn create(self: *EntyClass, reqdata: Value, ctrl: Value) OpResult {
    const ctx = self.utility.make_context(CtxSpec{
        .opname = "create",
        .ctrl = ctrl,
        .mtch = self.mtch,
        .data = self.data,
        .reqdata = reqdata,
    }, self.ent_ctx());
    return self.run_op(ctx, create_post_done);
}

fn create_post_done(self: *EntyClass, ctx: *Context) void {
    if (ctx.result) |result| {
        const resdata = result.resdata;
        if (!h.is_noval(resdata)) {
            const cm = h.to_map(h.clone(resdata));
            self.data = if (cm == .object) cm else h.omap();
        }
    }
}

// EJECT-END

File: ts/project/.sdk/src/cmp/rust/fragment/EntityCreateOp.fragment.rs
// EJECT-START

fn create(&self, reqdata: Value, ctrl: Value) -> Result<Value, ProjectNameError> {
    let ctx = self.utility.make_context(
        CtxSpec {
            opname: Some("create".to_string()),
            ctrl: Some(ctrl),
            mtch: Some(self.mtch.borrow().clone()),
            data: Some(self.data.borrow().clone()),
            reqdata: Some(reqdata),
            ..Default::default()
        },
        Some(&self.ent_ctx()),
    );

    self.run_op(&ctx, &|ctx| {
        if let Some(result) = ctx.result.borrow().clone() {
            let resdata = result.borrow().resdata.clone();
            if !resdata.is_noval() && !resdata.is_null() {
                *self.data.borrow_mut() = match to_map(&vs::clone(&resdata)) {
                    Value::Map(m) => Value::Map(m),
                    _ => Value::empty_map(),
                };
            }
        }
    })
}

// EJECT-END
